File: PlayWall/src/main/scala/de/tobias/playpad/action/actions/cart/handler/PlayHoldHandler.scala
package de.tobias.playpad.action.actions.cart.handler

import de.thecodelabs.midi.event.KeyEventType
import de.tobias.playpad.action.actions.CartAction
import de.tobias.playpad.pad.Pad

class PlayHoldHandler extends CartActionHandler {

	override def performAction(keyEventType: KeyEventType, cartAction: CartAction, pad: Pad): Unit = {
		if (keyEventType eq KeyEventType.DOWN) {
			if (pad.isReady) { // Allow the listener to send the feedback
				// cartAction.getPadPositionListener.setSend(false) TODO Fix
				pad.play()
			}
		} else if (keyEventType eq KeyEventType.UP) {
			if (pad.isPlay) {
				pad.stop()
			}
		}
	}
}

File: PlayWall/src/main/scala/de/tobias/playpad/action/actions/cart/handler/PlayHoldHandler.java
package de.tobias.playpad.action.actions.cart.handler;

import de.thecodelabs.midi.event.KeyEventType;
import de.tobias.playpad.action.actions.CartAction;
import de.tobias.playpad.pad.Pad;

public class PlayHoldHandler implements CartActionHandler
{
	@Override
	public void performAction(KeyEventType keyEventType, CartAction cartAction, Pad pad)
	{
		if(keyEventType.equals(KeyEventType.DOWN))
		{
			if(pad.isReady())
			{
				pad.play();
			}
		}
		else if(keyEventType.equals(KeyEventType.UP))
		{
			if(pad.isPlay())
			{
				pad.stop();
			}
		}
	}
}